modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/loaders/Common.scala

```diff
@@ -99,25 +96,34 @@ object Common {
     *
     * @param cliConfig RDB Loader app configuration
     */
-  def discover[F[_]: Monad: Timer: Cache: Logging: AWS](cliConfig: CliConfig): LoaderAction[F, List[DataDiscovery]] = {
+  def discover[F[_]: Monad: Timer: Cache: Logging: AWS](cliConfig: CliConfig): DiscoveryStream[F] = {
     // Shortcuts
     val shredJob = cliConfig.configYaml.storage.versions.rdbShredder
     val region = cliConfig.configYaml.aws.s3.region
     val assets = cliConfig.configYaml.aws.s3.buckets.jsonpathAssets
 
-    val target = cliConfig.folder match {
-      case Some(f) =>
-        DataDiscovery.InSpecificFolder(f)
+    cliConfig.target.messageQueue match {
+      case Some(name) =>
+        AWS[F]
+          .readSqs(name)
+          .translate(LoaderAction.fromFK[F])
+          .evalMap { s =>
+            LoaderMessage.fromString(s.data) match {
+              case Right(message: LoaderMessage.ShreddingComplete) =>
+                DataDiscovery.fromLoaderMessage[F](region, assets, message).map { discovery => (discovery, s.ack) }
+              case Left(error) =>
+                LoaderAction.liftE[F, DataDiscovery](LoaderError.LoaderLocalError(error).asLeft[DataDiscovery]).map { discovery => (discovery, s.ack) }
```

Comment:
If a SQS message fails to be parsed as a `LoaderMessage`, do we ack it and drop it ? We will need to put it in the manifest table.

Reply:
I think there are two scenarios:

1. Message is valid against its schema and decoded, but cannot be processed. I.e. JSONPath file cannot be discovered or S3 bucket doesn't exist. Or basically almost anything except "connection failure". In that case - yeah, we write it to the manifest, ack and notify Support. It will be easy to discover errors like this because they don't have attached row in `actions` table (another part of the manifest).
2. Message is invalid (what we have on this line). The problem with manifest here is that we simply cannot write it there because likely it has completely different structure, i.e. no expected timestamps, ids whatever. Maybe somebody sent rubbish to an SQS topic just for fun. In this case I think we must:

   1. Notify Support with high severity
   2. Log it as an error
   3. Ack - we have no choice

Luckily, I think these will almost never happen. Unless we restrict RDB Loader consumption too much (to `1-0-0`) and it will eventually fail to decode `1-0-1` after an upgrade.